lms/djangoapps/ccx/utils.py

```diff
@@ -24,3 +52,206 @@ def get_ccx_from_ccx_locator(course_id):
         )
         return None
     return ccx[0]
+
+
+def get_date(ccx, node, date_type=None, parent_node=None):
+    """
+    This returns override or master date for section, subsection or a unit.
+
+    :param ccx: ccx instance
+    :param node: chapter, subsection or unit
+    :param date_type: start or due
+    :param parent_node: parent of node
+    :return: start or due date
+    """
+    date = get_override_for_ccx(ccx, node, date_type, None)
+    if date_type == "start":
+        master_date = node.start
+    else:
+        master_date = node.due
+
+    if date is not None:
+        # Setting override date [start or due]
+        date = date.strftime('%Y-%m-%d %H:%M')
+    elif not parent_node and master_date is not None:
+        # Setting date from master course
+        date = master_date.strftime('%Y-%m-%d %H:%M')
+    elif parent_node is not None:
+        # Set parent date (vertical has same dates as subsections)
+        date = get_date(ccx, node=parent_node, date_type=date_type)
+
+    return date
+
+
+def validate_date(year, month, day, hour, minute):
+    """
+    avoid corrupting db if bad dates come in
+    """
+    valid = True
+    if year < 0:
+        valid = False
+    if month < 1 or month > 12:
+        valid = False
+    if day < 1 or day > 31:
+        valid = False
+    if hour < 0 or hour > 23:
+        valid = False
+    if minute < 0 or minute > 59:
+        valid = False
+    return valid
```

Comment:
Rather than returning a boolean, validation functions should raise an exception if the input is invalid. Return values can be ignored, but exceptions *must* be caught and handled, and an invalid result should never be ignored. (Django's `validate_email` function behaves like this, for example.)